front/pages/api/v1/w/[wId]/data_sources/[name]/databases/[id]/index.ts

```diff
@@ -0,0 +1,104 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { getDataSource } from "@app/lib/api/data_sources";
+import { Authenticator, getAPIKey } from "@app/lib/auth";
+import { CoreAPI, CoreAPIDatabase } from "@app/lib/core_api";
+import { isDevelopmentOrDustWorkspace } from "@app/lib/development";
+import logger from "@app/logger/logger";
+import { apiError, withLogging } from "@app/logger/withlogging";
+
+type GetDatabaseResponseBody = {
+  database: CoreAPIDatabase;
+};
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<GetDatabaseResponseBody>
+): Promise<void> {
+  const keyRes = await getAPIKey(req);
+  if (keyRes.isErr()) {
+    return apiError(req, res, keyRes.error);
+  }
+
+  const { auth } = await Authenticator.fromKey(
+    keyRes.value,
+    req.query.wId as string
+  );
+
+  const owner = auth.workspace();
+  const plan = auth.plan();
+  if (!owner || !plan) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace you requested was not found.",
+      },
+    });
+  }
+
+  if (!isDevelopmentOrDustWorkspace(owner)) {
+    res.status(404).end();
+    return;
+  }
+
+  const dataSource = await getDataSource(auth, req.query.name as string);
+  if (!dataSource) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "data_source_not_found",
+        message: "The data source you requested was not found.",
+      },
+    });
+  }
+
+  const databaseId = req.query.id;
+  if (!databaseId || typeof databaseId !== "string") {
+    return apiError(req, res, {
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: "The database id is missing.",
+      },
+    });
+  }
+
+  switch (req.method) {
+    case "GET":
```

Comment:
we don't check for builder status in other v1 endpoints?

Reply:
Api key means builder no?

Reply:
For now but maybe not tomorrow

Reply:
#defense_in_depth

Reply:
We don't do it anywhere else, but if you want I can change all the v1 endpoints to add the check ?

Reply:
Most likely preferrable as a separate PR though, wdyt ?

Reply:
If we don’t do it anywhere else then fine to keep as is 👍 as we’ll know we’ll have to do it when the time comes